### PR TITLE
NEO-2175: Select closing bug fix

### PR DESCRIPTION
**Pull request by enrique-prado**
[Link to Select Searchable story](https://deploy-preview-513--neo-react-library-storybook.netlify.app/?path=/story/components-select--searchable)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-devs`

This PR fixes a bug in both the Single Searchable & Multi Searchable select components that happens when a user clicks on it to open it and it closes itself.  The dropdown of options now remains open as expected.

- The bug was caused by the underlying onFocus implementation of downshift.
- Updated an existing test with a built in delay to make sure the select component remains open.

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved focus handling in the MultiSelect and SingleSelect components to prevent unintended actions when interacting with dropdowns.
	- Added functionality to close the dropdown when the input field gains focus, enhancing user experience.

- **Tests**
	- Enhanced test reliability for the Select component by implementing asynchronous checks for the `aria-expanded` property, ensuring accurate behavior during interactions.

### CODE REVIEW

**Review by yangAtSpoken — CHANGES_REQUESTED** [review]
Both searchable selects [here](https://neo-react-library-storybook.netlify.app/?path=/story/components-select--searchable) used to open on getting keyboard focus. 
 Now, this behavior is gone.  Is it possible to get it back?

File: src/components/Select/InternalComponents/MultiSelectSearchable.tsx [hunk 1]
```diff
@@ -43,6 +43,10 @@ export const MultiSelectSearchable = () => {
 		getToggleButtonProps();
 	const { id, onKeyDown, ...restInputProps } = getInputProps();
 
+	if (restInputProps.onFocus) {
+		restInputProps.onFocus = undefined; // Override downshift behavior to fix NEO-2175
+	}
```
Comment (joe-s-avaya) [inline src/components/Select/InternalComponents/MultiSelectSearchable.tsx:48]: nit: prefer one liner:
```suggestion
	delete restInputProps.onFocus; // remove downshift behavior to fix NEO-2175
```

File: src/components/Select/Searchable.test.jsx [hunk 2]
```diff
@@ -49,15 +50,19 @@ describe("Select", () => {
...
 				"span.neo-multiselect-combo__header",
 			);
 			const searchableElement = getByRole("combobox");
 			expect(searchableElement).toHaveAttribute("aria-expanded", "false");
 			fireEvent.click(toggleButton);
-			expect(searchableElement).toHaveAttribute("aria-expanded", "true");
+			await vi.waitUntil(
+				() =>
+					expect(searchableElement).toHaveAttribute("aria-expanded", "true"),
+				{ timeout: 500 },
```
Comment (joe-s-avaya) [inline src/components/Select/Searchable.test.jsx:64]: you shouldn't need a timeout here, please remove and test

File: src/components/Select/InternalComponents/MultiSelectSearchable.tsx (hunk 1 above)
Comment (joe-s-avaya) [reply src/components/Select/InternalComponents/MultiSelectSearchable.tsx:48]: same in `SingleSelectSearchable.tsx`

Comment (enrique-prado) [reply src/components/Select/InternalComponents/MultiSelectSearchable.tsx:48]: yes, that was my original implementation but biome flagged using delete as a performance issue and suggested I do the assignment to undefined. Thoughts?

Comment (yangAtSpoken) [reply src/components/Select/InternalComponents/MultiSelectSearchable.tsx:48]: [This codesandbox](https://codesandbox.io/p/sandbox/usecombobox-action-props-z6qkg?file=%2Fsrc%2Findex.js)  shows how to override onFocus().

File: src/components/Select/Searchable.test.jsx (hunk 2 above)
Comment (enrique-prado) [reply src/components/Select/Searchable.test.jsx]: removed timeout. It passes.

**Comment by yangAtSpoken** [comment]
Not sure if you have seen [this](https://github.com/downshift-js/downshift/issues/1095), might be helpful.

File: src/components/Select/InternalComponents/MultiSelectSearchable.tsx (hunk 1 above)
Comment (enrique-prado) [reply src/components/Select/InternalComponents/MultiSelectSearchable.tsx:48]: @yangAtSpoken @joe-s-avaya  Yes Yang, that would be another way of doing it.  In our case we don't want to pass a new implementation but simply delete the one from downshift.  Using the codesandbox pattern my code would look something like this:
onFocus: () => undefined
or 
onFocus: () => {}

Any thoughts?

Comment (yangAtSpoken) [reply src/components/Select/InternalComponents/MultiSelectSearchable.tsx:48]: Does it help if you call openMenu() in onFocus?  I would like to keep the current behavior when it gets the keyboard focus.

**Review by joe-s-avaya — APPROVED** [review]
there are a few things left to do, but you're aware of them so I'm approving ahead of your updates so that I don't block you

File: src/components/Select/InternalComponents/MultiSelectSearchable.tsx (hunk 1 above)
Comment (enrique-prado) [reply src/components/Select/InternalComponents/MultiSelectSearchable.tsx:48]: > the latter (`onFocus: () => {}`)

using the empty object causes a warning so I'll leave it as undefined.

Comment (enrique-prado) [reply src/components/Select/InternalComponents/MultiSelectSearchable.tsx:48]: > Does it help if you call openMenu() in onFocus? I would like to keep the current behavior when it gets the keyboard focus.

I tried the openMenu() solution but it does not fix the bug. So far overriding with undefined seems to be the best solution.

**Comment by enrique-prado** [comment]
> Not sure if you have seen [this](https://github.com/downshift-js/downshift/issues/1095), might be helpful.

I tried the openMenu trick but it didn't work.

File: src/components/Select/InternalComponents/MultiSelectSearchable.tsx [hunk 3]
```diff
@@ -43,6 +43,10 @@ export const MultiSelectSearchable = () => {
 		getToggleButtonProps();
 	const { id, onKeyDown, ...restInputProps } = getInputProps();
```
Comment (yangAtSpoken) [inline src/components/Select/InternalComponents/MultiSelectSearchable.tsx:44]:
```suggestion
	const { id, onKeyDown, ...restInputProps } = getInputProps({ onFocus: undefined });
```
This works for me on my local. It preserves the keyboard behavior.  Please verify.

Comment (yangAtSpoken) [reply src/components/Select/InternalComponents/MultiSelectSearchable.tsx:44]: Unfortunately, this can not be reproduced.

**Review by yangAtSpoken — APPROVED** [review]

**Pull request merged**